openpype/hosts/houdini/plugins/create/create_pointcache.py

```diff
@@ -37,14 +37,44 @@ def create(self, subset_name, instance_data, pre_create_data):
         }
 
         if self.selected_nodes:
-            parms["sop_path"] = self.selected_nodes[0].path()
+            parent = self.selected_nodes[0]
+            parms["sop_path"] = parent.path()
+
+            child_render = ""
+            child_output = ""
 
             # try to find output node
-            for child in self.selected_nodes[0].children():
+            for child in parent.children():
+                if child.isGenericFlagSet(hou.nodeFlag.Render):
+                    child_render = child
                 if child.type().name() == "output":
-                    parms["sop_path"] = child.path()
+                    child_output = child
                     break
 
+            # create output node if not exists
+            if not child_output:
+                child_output = parent.createNode("output", "OUTPUT")
+                child_output.setFirstInput(child_render)
+
+                child_output.setDisplayFlag(1)
+                child_output.setRenderFlag(1)
+                child_output.moveToGoodPosition()
+
+            paths = child_output.geometry().findPrimAttrib("path") and \
+                      child_output.geometry().primStringAttribValues("path")
+
+            # Create default path value if missing
+            if not paths:
+                path_node = parent.createNode("name", "AUTO_PATH")
+                path_node.parm("attribname").set("path")
+                path_node.parm("name1").set('`opname("..")`/`opoutput(".", 0)`') # noqa
+
+                path_node.setFirstInput(child_output.input(0))
+                path_node.moveToGoodPosition()
+                child_output.setFirstInput(path_node)
+                child_output.moveToGoodPosition()
+
+            parms["sop_path"] = child_output.path()
```

Comment:
I still don't think this is a good idea to **always** do on create.

It doesn't **require** an output node.
It doesn't **require** the path parameter to be `path` but just to be set. The user could still customize the `path` attribute (the name of it on the rop node) and have it publish fine as far as I know.

Since it's not a requirement it feels quite dirty to forcefully throw down these extra nodes all the time, especially also because deleting the publish instance does not delete these nodes.

I'd argue this is not the way to go unfortunately.

As such, I'd vote for not doing this on create.

Reply:
Let me know know if I got it right : Do you think that implementing the repair action is sufficient ?

Reply:
I agree with @BigRoy that he implementation of path parameter can be just in the repair function. User can add/customize the path attributes with some other different nodes knowing that there is not only name node allowing to add more attributes.
But I think the output node can be implemented when the user uses selection if there isn't one.

Reply:
So, it's better to add `output node` if missing on creation 

And, repairing path attribute if missing on publishing

Reply:
> when the user uses selection if there isn't one.

But if there's no selection - then _where_ do you add it?
Note that the tab menu creation for OpenPype creators in the network editor in Houdini actually already does exactly that.